cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/UninstallUtils.java

```diff
@@ -0,0 +1,120 @@
+        String packageName;
+        String version;
+        // Get org-name
+        int orgNameIndex = fullPkgPath.indexOf("/");
+        if (orgNameIndex != -1) {
+            orgName = fullPkgPath.substring(0, orgNameIndex);
+        } else {
+            throw new BLangCompilerException("no org-name is provided");
+        }
+        // Get package name
+        int packageNameIndex = fullPkgPath.indexOf(":");
+        if (packageNameIndex != -1) { // version is provided
+            packageName = fullPkgPath.substring(orgNameIndex + 1, packageNameIndex);
+            version = fullPkgPath.substring(packageNameIndex + 1, fullPkgPath.length());
+        } else {
+            throw new BLangCompilerException("no package version is provided");
+        }
+        Path homeRepoPath = BALLERINA_HOME_PATH.resolve(ProjectDirConstants.DOT_BALLERINA_REPO_DIR_NAME);
+        Path cacheRepoPath = BALLERINA_HOME_PATH.resolve(ProjectDirConstants.CACHES_DIR_NAME)
+                                                .resolve(ProjectDirConstants.BALLERINA_CENTRAL_DIR_NAME);
+        Path pkgPath = Paths.get(orgName, packageName, version);
+
+        // Check if package is installed locally
+        if (Files.exists(homeRepoPath.resolve(pkgPath), LinkOption.NOFOLLOW_LINKS)) {
+            deletePackage(homeRepoPath, pkgPath, fullPkgPath, packageName);
+        } else if (Files.exists(cacheRepoPath.resolve(pkgPath), LinkOption.NOFOLLOW_LINKS)) {
+            deletePackage(cacheRepoPath, pkgPath, fullPkgPath, packageName);
+        } else {
+            // package to be uninstalled doesn't exists
+            throw new BLangCompilerException("package does not exist " + fullPkgPath);
+        }
+    }
+
+    /**
+     * Remove package from home repository.
+     *
+     * @param repoPath    path of the home repository
+     * @param pkgPath     package path
+     * @param fullPkgPath full package path as given by user
+     * @param pkgName     package name
+     */
+    private static void deletePackage(Path repoPath, Path pkgPath, String fullPkgPath, String pkgName) {
+        try {
+            Path path = repoPath.resolve(pkgPath);
+            // Delete the package zip
+            Files.deleteIfExists(path.resolve(pkgName + ProjectDirConstants.BLANG_COMPILED_PKG_EXT));
+
+            // Delete the empty directories
+            deleteEmptyDirsUpTo(path, repoPath);
+
+            // Print that the package was successfully uninstalled
+            outStream.println(fullPkgPath + " successfully uninstalled");
+        } catch (IOException e) {
+            throw new BLangCompilerException("error uninstalling package " + fullPkgPath);
+        }
+    }
+
+    /**
+     * Delete empty directories.
+     *
+     * @param from package directory path
+     * @param to   home repository path
+     * @throws IOException throw an exception if an error occurs
+     */
+    private static void deleteEmptyDirsUpTo(Path from, Path to) throws IOException {
```

Comment:
This method is not clear. Why do we need the second arg "to" ?

Reply:
After deleting the package.zip from the home repository we should delete the empty directories as well. We need to delete all folders from the package path to the home directory only if they are empty (going backwards).

Reply:
Changed the method name to  'deleteEmptyParentDirs' with e74db59